independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/PublisherResponseHandler.java

```diff
@@ -63,67 +61,43 @@ public void onNext(Object item) {
             } else {
                 event = new OutboundSseEventImpl.BuilderImpl().data(item).build();
             }
-            SseUtil.send(requestContext, event, staticCustomizers).whenComplete(new BiConsumer<Object, Throwable>() {
-                @Override
-                public void accept(Object v, Throwable t) {
-                    if (t != null) {
-                        // need to cancel because the exception didn't come from the Multi
-                        subscription.cancel();
-                        handleException(requestContext, t);
-                    } else {
-                        // send in the next item
-                        subscription.request(1);
-                    }
+            SseUtil.send(requestContext, event, staticCustomizers).whenComplete((v, t) -> {
+                if (t != null) {
+                    // need to cancel because the exception didn't come from the Multi
+                    subscription.cancel();
+                    handleException(requestContext, t);
+                } else {
+                    // send in the next item
+                    subscription.request(demand);
                 }
             });
         }
     }
 
-    private static class ChunkedStreamingMultiSubscriber extends StreamingMultiSubscriber {
```

Comment:
Removed this inner class, because it's doing what `StreamingMultiSubscriber` does with `encodeAsJsonArray==false`